PracticeGit/Test.c

```diff
@@ -1,13 +1,151 @@
 // ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
-// ------------------------------------------------------------------
-// test.c
 // ------------------------------------------------------------------------------------------------
+// test.c -  Program that Executes test cases and checks the user input for sort & search.
+// ------------------------------------------------------------------------------------------------
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+#define MAXSIZE 10
+#define ELEMENT_NOT_FOUND -1
+#define RED     "\033[1;31m"
+#define GREEN   "\033[1;32m"
+#define RESET   "\033[0m"
+
+/// <summary> Prints the elements of an array. </summary>
+void PrintArray (int array[], int size);
+
+/// <summary> Runs test cases for the QuickSort function. </summary>
+void RunQuickSortTest ();
+
+/// <summary> Gets user input for an array, sorts it, and performs a binary search. </summary>
+void GetUserInput ();
+
+/// <summary>Compare Arrays returns true or false. </summary>
+bool ArraysEqual (int arr1[], int arr2[], int size);
+
+typedef struct {
+   int array[MAXSIZE], size;
+} TestCase;
+
+void PrintArray (int array[], int size) {
+   for (int i = 0; i < size; i++) printf ("%3d", array[i]);
+}
+
+bool ArraysEqual (int arr1[], int arr2[], int size) {
+   for (int i = 0; i < size; i++) {
+      if (arr1[i] != arr2[i]) return false;
```

Comment:
When checking if the arrays are equal, the first thing we must do is to check whether both the arrays are of the same size. And passing the size of the arrays as parameter is not a good idea.